### ISSUE TITLE
Tracking issue: new home for juju charm related work

### ISSUE THREAD

**Issue by kwmonroe** [kubernetes/community#2892]
In reference to https://github.com/kubernetes/kubernetes/pull/70430, we're looking for an appropriate home for code that lives in the deprecated `kubernetes/cluster/juju` tree:

https://github.com/kubernetes/kubernetes/tree/master/cluster/juju

We're leaning towards requesting a new `charms` repo in the kubernetes org; kubernetes-sigs was also suggested as a potential parent org. Questions/comments related to this move would be appreciated.

/cc @roberthbailey

**Comment by kwmonroe** [comment]
@kubernetes/sig-cluster-lifecycle-proposals

**Comment by timothysc** [comment]
This should really live out of tree, perhaps in kubernetes-sigs, but I'm wondering how much usage does this tool actually get.?.?  

Typically juju's are only used by canonical, and not more broadly applicable.

**Comment by kwmonroe** [comment]
> Typically juju's are only used by canonical, and not more broadly applicable.

There's a broad Juju following both in- and outside of Canonical.  Juju excels at managing K8s across clouds -- from AWS to VMware to home-grown metal -- as well as enabling out-of-cluster tool integration with K8s (Keystone auth, Prometheus metrics, etc).

**Comment by luxas** [comment]
I'm okay with / supportive of creating a new kubernetes-sigs repo (e.g. `kubernetes-sigs/juju-charms` or `kubernetes-sigs/cluster-api-{juju,charms}`) for getting this code out of the core k8s repo. It shouldn't be in the core repo for multiple reasons.

> a broad

Are you able to mention confirmed numbers or public users (optional, but it'd help build your case. You can do this to the leads privately as well if you want)?

One requirement for me to keep this inside of the "core sphere of k8s" (vs move this to your own canonical org) is that you submit conformance tests. To me it seems like you haven't done this. Is that correct?
The second requirement is that you start working to submit conformance results periodically (~every day at least) to the k8s testgrid (http://testgrid.k8s.io)
Thirdly, I'd strongly argue for that you embed kubeadm in your cluster creation tool. Many other providers have done this (minikube, kubespray, many Cluster API providers, etc.), and you'd benefit too (along with the ecosystem).
If/when SIG Cluster Lifecycle sponsors this code in kubernetes-sigs, you should also periodically check in with status updates for this subproject to our SIG meeting biweekly on Tuesdays FYI.
Finally, have you considered adopting / supporting the Cluster API spec for cluster creation? That'd help both you and us be on the same page.

I'm just trying to be clear with the expectations here. We have two options going forward, move this code to a new repo in `kubernetes-sigs` or your own Github organization. If we do keep it as a `kubernetes-sigs` project, there are community-wide expectations on how the project shall function and be governed. If you accept these terms, as roughly highlighted above, we're happy to create the new repo. To date this code seems to have flewn under everybody's radar, with no SIG owning it. So although we've had several policies (code in the core repo or docs must be proven conformant), this hasn't got the attention it needs because of governance missing.

Let's fix this by moving the code to either new organization.
What do you think is the best option here @kwmonroe?
Thank you for creating this issue!

**Comment by roberthbailey** [comment]
Thanks for following up on this @luxas and bringing it back onto our radar for 2019. 

I agree that if we plan to keep this code in the Kubernetes community repositories then the maintenance model will need to change and we need to adhere to the requirements for any SIG / SIG subproject. 

One comment about this quote from the PR comment thread:

> What's worked well for other projects like ours (kops, kubernetes-anywhere, etc) is not being in kubernetes/kubernetes but in kubernetes/project so I recommend for consistency's sake we just move to kubernetes/charms. There aren't projects like ours in kubernetes-sigs so it would seem odd to go that route unless and until we move the other projects like ours there as well.

kops and minikube are in the kubernetes org because those repositories was created before kubernetes-sigs existed. kubernetes-anywhere is deprecated and rather than spend time moving it to kubernetes-sigs we are trying to remove our reliance on it and move it to the graveyard. [kubespray](https://github.com/kubernetes-sigs/kubespray) has moved to kubernetes-sigs and ideally we've move kops and minikube there as well. So I think there is ample precedent for living in that location, assuming proper SIG sponsorship, rather than the kubernetes org.

**Comment by kwmonroe** [comment]
@luxas and @roberthbailey, thanks for the followup, and I apologize for the delayed response. If you'll recall, our biggest issue with being in-tree is that charm source doesn't necessarily follow upstream k8s releases; we've had to work around this by maintaining and releasing charms from a fork.

Fork management and sync has not been ideal, so we've investigated both SIG and Canonical org ownership of charm code. Based on info from cluster-lifecycle and the comments here, we've decided that k8s charm source would fit best under the Canonical org. We're working to evacuate in-tree ./cluster/juju and move to a new repo. The last thing we want to do is move k8s charm source twice, so we'll need a bit of time to make sure documentation, CI, and related links are pointing at the right location.

Thanks for the patience as we re-home this source. I'll close this issue with the new ./cluster/juju location asap.
